### PR TITLE
chore: Replace dependency on unexported otelcol functions

### PR DESCRIPTION
**Pull request by dehaansa**
#### PR Description
Inspired by https://github.com/grafana/alloy/pull/1023 - replacing the call to an unexported function to utilizing the intended functionality from otelcol.

- [X] Tests updated
- [X] Config converters updated

### CODE REVIEW

**Review by ptodev — COMMENTED** [review]
Amazing, thank you! I didn't realise that it's possible to do this 😆  I thought the public OTel functions only allow generating the config from a file.

File: internal/static/traces/config_test.go [hunk 1]
```diff
@@ -1172,6 +1172,7 @@ extensions:
   oauth2client/otlphttp0:
     client_id: someclientid
     client_secret: someclientsecret
+    endpoint_params: {}
```
[the remark this answers is not in the document]
Comment (dehaansa) [reply internal/static/traces/config_test.go:1175]: It's a symptom of the way that the extra yaml marshal back to []byte to be used in the URL handles the lack of endpoint_params/endpoint_params = nil. We can try to continue to modify the encoder used in the config, but as this wasn't a change that would impact behavior I didn't try to fix it.

I'll jump back in and take a look at this again later today to see if I can resolve.

[the remark this answers is not in the document]
Comment (dehaansa) [reply internal/static/traces/config_test.go]: This appears to be a symptom of the way the code uses a raw map[string]interface{} for an otelconfig, under which you lose the `omitEmpty` context at the top level for the url.Values object.

As this ToOtelConfig pattern is no longer necessary with the correct otel loading pattern, I removed these ToOtelConfig patterns and this was resolved, as well as removing incorrect/unnecessary tls & tls.cipher_suites blocks). See the [commit](https://github.com/grafana/alloy/pull/2148/commits/3cebba7fa3b57c38efc6e572fc7015352f677e35)

[the remark this answers is not in the document]
Comment (dehaansa) [reply internal/static/traces/config_test.go]: Yes, the attributes need to match identically, however the ToOtelConfig pattern I meant was just used in two places, SSLConfig and OAuthConfig, and they were 1. only defined in this file and 2. used identical keys.

**Review by ptodev — APPROVED** [review]
Thank you! Would you like to also do the same change to the Agent?
We won't be able to rid of the fork entirely, but it'll help - there will be less things to modify on the fork when we upgrade it.

**Comment by wildum** [comment]
@ptodev I don't believe that we are planning on updating the Otel version for the Agent

**Comment by ptodev** [comment]
I don't insist on the change being done on the Agent. I'll leave it up to @dehaansa to decide if it's worth his effort. Maybe it's too much risk for too little benefit.

**Comment by wildum** [comment]
We already use a custom version of the fork for the agent because of some CVEs (see https://github.com/grafana/agent/blob/main/go.mod#L777) so to get rid of the fork we would either need to copy some more code or we would need to upgrade the Otel version at the same time to get the CVEs fix.
I personally don't think that's worth the effort

**Pull request merged**